### PR TITLE
Filters - mobile view

### PR DESCRIPTION
**Pull request by Adamik10**
#### Link to issue
https://reload.atlassian.net/browse/DDFBRA-127

#### Description
Introduces mobile filters + some small styling changes.

#### Screenshot of the result
-

#### Additional comments or questions
-

### CODE REVIEW

File: components/shared/badge/BadgeButton.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+const BadgeButton = ({ onClick, isActive = false, classNames, children }: BadgeButtonProps) => {
+  return (
+    <button
+      onClick={onClick}
+      className={cn([
+        `focus-visible h-[29px] w-auto self-start whitespace-nowrap rounded-full bg-background-overlay px-4
+        py-2 text-typo-caption hover:animate-wiggle`,
+        isActive && "bg-foreground text-background",
+        classNames,
+      ])}>
```
Comment (ThomasGross) [inline components/shared/badge/BadgeButton.tsx:21]: You dont need an array here

```suggestion
      className={cn(
        `focus-visible h-[29px] w-auto self-start whitespace-nowrap rounded-full bg-background-overlay px-4
        py-2 text-typo-caption hover:animate-wiggle`,
        isActive && "bg-foreground text-background",
        classNames,
      )}>
```

File: components/shared/searchFilters/SearchFiltersColumn.tsx [hunk 2]
```diff
@@ -44,48 +45,38 @@ const SearchFiltersColumn = ({
     <>
       <div
         key={facet.name}
-        className={cn(["relative", !isLast && "min-w-32 flex-1", isLast && "flex-2"])}>
-        <h3 className="mb-2 text-typo-caption uppercase">{getFacetTranslation(facetFilter)}</h3>
+        className={cn(["relative ml-[-4px]", !isLast && "min-w-32 flex-1", isLast && "flex-2"])}>
```
Comment (ThomasGross) [inline components/shared/searchFilters/SearchFiltersColumn.tsx:48]:
```suggestion
        className={cn("relative ml-[-4px]", !isLast && "min-w-32 flex-1", isLast && "flex-2")}>
```

File: lib/config/resolvers/search.ts [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import { FacetFieldEnum } from "@/lib/graphql/generated/fbi/graphql"
 
 const search = {
-  "search.item.limit": 9,
+  "search.item.limit": 18,
```
Comment (ThomasGross) [inline lib/config/resolvers/search.ts:4]: 9 means faster load times also serverside. Could be nice to keep. Maybe 12 could be the sweetspot

```suggestion
  "search.item.limit": 9,
```

File: components/shared/searchFilters/SearchFiltersColumn.tsx [hunk 4]
```diff
@@ -44,48 +45,38 @@ const SearchFiltersColumn = ({
...
-                bg-background-overlay pl-2 pr-4 text-typo-caption hover:animate-wiggle`,
-                isExpanded && "mt-1"
-              )}>
-              <Icon className={cn("h-8 w-8", isExpanded && "rotate-180")} name="arrow-down" />
-              <p>
-                {!isExpanded && "Flere"} {isExpanded && "Skjul"}
-              </p>
-            </button>
-          </div>
+            <Icon className={cn("h-8 w-8", isExpanded && "rotate-180")} name="arrow-down" />
```
Comment (ThomasGross) [inline components/shared/searchFilters/SearchFiltersColumn.tsx:75]: Im pretty sure "isExpanded" will end up in the class if you're using this time of if statement

```suggestion
            <Icon className={cn("h-8 w-8", isExpanded ? "rotate-180" : '')} name="arrow-down" />
```

File: components/shared/searchFilters/SearchFiltersColumn.tsx [hunk 5]
```diff
@@ -44,48 +45,38 @@ const SearchFiltersColumn = ({
...
       <div
         key={facet.name}
-        className={cn(["relative", !isLast && "min-w-32 flex-1", isLast && "flex-2"])}>
-        <h3 className="mb-2 text-typo-caption uppercase">{getFacetTranslation(facetFilter)}</h3>
+        className={cn(["relative ml-[-4px]", !isLast && "min-w-32 flex-1", isLast && "flex-2"])}>
+        <h3 className="mb-2 pl-2 text-typo-caption uppercase">
+          {getFacetTranslation(facetFilter)}
+        </h3>
         <div
           className={cn([
```
Comment (ThomasGross) [inline components/shared/searchFilters/SearchFiltersColumn.tsx:53]: remove the extra array

```suggestion
          className={cn(
```

File: components/shared/searchFilters/SearchFiltersColumn.tsx [hunk 6]
```diff
@@ -4,6 +4,7 @@ import React, { useEffect, useRef, useState } from "react"
 import { SearchFacetFragment, SearchFiltersInput } from "@/lib/graphql/generated/fbi/graphql"
 import { cn } from "@/lib/helpers/helper.cn"
 
+import BadgeButton from "../badge/BadgeButton"
```
Comment (ThomasGross) [inline components/shared/searchFilters/SearchFiltersColumn.tsx:7]: routes from root

File: components/shared/searchFilters/SearchFiltersColumn.tsx [hunk 7]
```diff
@@ -4,6 +4,7 @@ import React, { useEffect, useRef, useState } from "react"
...
 import Icon from "../icon/Icon"
```
Comment (ThomasGross) [inline components/shared/searchFilters/SearchFiltersColumn.tsx:8]: routes from root

File: components.json [hunk 8]
```diff
@@ -12,7 +12,7 @@
   },
   "aliases": {
     "components": "@/components/shadcn",
-    "utils": "@/lib/shadcn/utils",
+    "utils": "@/lib/helpers",
```
Comment (ThomasGross) [inline components.json:15]: I think this should be changed back 

```suggestion
    "utils": "@/lib/shadcn/utils",
```

File: components/shared/searchFilters/SearchFiltersMobile.tsx [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+
+import {
+  Accordion,
+  AccordionContent,
+  AccordionItem,
+  AccordionTrigger,
+} from "@/components/accordion/Accordion"
+import { SearchFacetFragment, SearchFiltersInput } from "@/lib/graphql/generated/fbi/graphql"
+
+import BadgeButton from "../badge/BadgeButton"
```
Comment (ThomasGross) [inline components/shared/searchFilters/SearchFiltersMobile.tsx:12]: import from root

File: components/shared/searchFilters/SearchFiltersMobile.tsx [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+import Icon from "../icon/Icon"
```
Comment (ThomasGross) [inline components/shared/searchFilters/SearchFiltersMobile.tsx:13]: import from root

File: components/shared/searchFilters/SearchFiltersMobile.tsx [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+import { Sheet, SheetContent, SheetTrigger } from "../sheet/Sheet"
```
Comment (ThomasGross) [inline components/shared/searchFilters/SearchFiltersMobile.tsx:14]: import from root

**Review by ThomasGross — COMMENTED** [review]
Some minor things, overall great!

File: lib/config/resolvers/search.ts (hunk 3 above)
Comment (Adamik10) [reply lib/config/resolvers/search.ts:4]: If the results are non-divideable by both 2 and 3 then we end up with something like this:
image

**Review by ThomasGross — APPROVED** [review]

**Pull request merged**